src/main/java/decodes/cwms/CwmsSqlDatabaseIO.java

```diff
@@ -323,97 +337,34 @@ public IntervalDAI makeIntervalDAO()
 	public Connection getConnection()
 	{
 		// Called from DAOs and DbIo to get a new connection from the pool.
-		if (conInfo == null || conInfo.getLoginInfo() == null)
+		if (conInfo == null || conInfo.getLoginInfo() == null || pool == null)
 		{
 			Logger.instance().failure(module + ".getConnection -- loginInfo is null! DB not initialized?");
 			return null;
```

Comment:
Eventually I'd like to see this behavior changed to throwing a runtime exception. Trying to setup isolated tests becomes difficult when I'm getting null pointer exceptions on null connections.

Reply:
That's a good point, there's not really any recovery upstream from this anyways.